scripts/exglobal_makeprepbufr.sh

```diff
@@ -553,44 +553,52 @@ fi  # end [ "$DO_QC" = 'YES' ]
 ## create combined ocean data dump file expected by NSST
 if [[ "$MAKE_NSSTBUFR" == 'YES' ]]; then
    > nsstbufr
-   chgrp rstprod nsstbufr
-   err_ch=$?
-   if [ $err_ch -eq 0 ]; then
+   chmod 664 nsstbufr
 
-      DTYPS_nsst='sfcshp tesac bathy trkob subpfl saldrn'
+   DTYPS_nsst='sfcshp tesac bathy trkob subpfl saldrn'
 
-      echo "xglm: DTYPS_nsst='$DTYPS_nsst'"
+   echo "xglm: DTYPS_nsst='$DTYPS_nsst'"
+    
+   for type in $DTYPS_nsst ; do
+      if [ -f ${tstsp}$type.$tmmark.bufr_d ]; then
+         file=${tstsp}$type.$tmmark.bufr_d
+      elif [ -s ${COMSP}$type.$tmmark.bufr_d ]; then
+         file=${COMSP}$type.$tmmark.bufr_d
+      fi
 
-      for type in $DTYPS_nsst ; do
-         if [ -f ${tstsp}$type.$tmmark.bufr_d ]; then
-            file=${tstsp}$type.$tmmark.bufr_d
-         elif [ -s ${COMSP}$type.$tmmark.bufr_d ]; then
-            file=${COMSP}$type.$tmmark.bufr_d
+      if [ -s $file ]; then
+           cat $file >> nsstbufr
+           err_cat=$?
+        if [ $err_cat -ne 0 ]; then
+           msg="**WARNING: exit status $err_cat from cat of $file to nsstbufr"
+           $DATA/postmsg "$jlogfile" "$msg"
          fi
+      else
+         echo $file is empty or does not exist
+      fi # [ -s $file ]
 
-         if [ -s $file ]; then
-              cat $file >> nsstbufr
-              err_cat=$?
-            if [ $err_cat -ne 0 ]; then
-              msg="**WARNING: exit status $err_cat from cat of $file to nsstbufr"
-              $DATA/postmsg "$jlogfile" "$msg"
-            fi
-         else
-            echo $file is empty or does not exist
-         fi # [ -s $file ]
-
-      done # for type in $DTYPS_nsst 
+   done # for type in $DTYPS_nsst
+   
+   cp nsstbufr $COMOUT/${RUN}.${cycle}.nsstbufr
+   chmod 664 $COMOUT/${RUN}.${cycle}.nsstbufr
 
-      cp nsstbufr $COMOUT/${RUN}.${cycle}.nsstbufr
+   if [ "$CHGRP_RSTPROD" = 'YES' ]; then
+      chgrp rstprod nsstbufr
       chgrp rstprod $COMOUT/${RUN}.${cycle}.nsstbufr
```

Comment:
Remove duplicate and keep just the second line:
```suggestion
      chgrp rstprod $COMOUT/${RUN}.${cycle}.nsstbufr
```

Reply:
@KateFriedman-NOAA I wasn't sure how to handle that since `nsstbufr` is within the stmp directory before being copied to `$COMOUT/${RUN}.${cycle}.nsstbufr`. Will commit these suggested changes. Are the files in the stmp directory always removed?

Reply:
If KEEPDATA=NO the stmp directory will always be removed, that's the default setting in production.